### DISCUSSION TITLE
Migrating from 3.3.10 to 3.4.4 and got similar error:

### DISCUSSION THREAD

**Discussion by sincomil** [netbox-community/netbox#11778] [Help Wanted!]
Migrating from 3.3.10 to 3.4.4 and got similar error:

```
Running migrations:
  Applying tenancy.0008_unique_constraints...Traceback (most recent call last):
  File "/opt/netbox-git/venv/lib/python3.9/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
psycopg2.errors.UniqueViolation: could not create unique index "tenancy_contactassignment_unique_object_contact_role"
DETAIL: Key (content_type_id, object_id, contact_id, role_id)=(87, 1499, 4, 2) is duplicated.
```

_Originally posted by @sincomil in https://github.com/netbox-community/netbox/issues/10956#issuecomment-1434577676_

**Comment by kkthxbye-code** [comment]
This is more suited for a discussion. Issues require that the template is followed. I'll convert it to a discussion for you.

**Comment by sincomil** [comment]
previous [issue](https://github.com/netbox-community/netbox/issues/10956#event-7834964442) was marked as bug and then was closed. but I think this bug is still there. Nothing was fixed in 3.4 as was planned

**Reply by zmaster7** [reply]
As @jeremystretch mentioned in the original issue, the intended behavior was to not allow duplicate contacts with the same role, but different priorities on the same object. In 3.4, the behavior is working as intended, so it was 3.3 that technically had the bug. I had originally opened the issue as I wasn't sure which was the intended outcome.

In my case, the resolution was to simply remove the "secondary" priority and leave the "primary" priority in place for duplicate contact/roles. I couldn't think of a valid reason to keep both a "primary" and "secondary" priority listing under the same role for an assignment to argue against the intended behavior, as in my case the contact got assigned twice programmatically after another person had left the org.

**Reply by zmaster7** [reply]
On a side note, content_type_id = 87 is for the 'adminuser' model, at least in v3.4. If that is the same id for v3.3 I'm not sure how that is possible? I don't see any way to assign contacts to that model.
